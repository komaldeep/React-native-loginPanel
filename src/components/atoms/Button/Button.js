import React, {Component} from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import multipleStyles from 'react-native-multiple-styles'
const Dimensions = require('Dimensions');
import PropTypes from 'prop-types';

const viewport = {
  fullWidth: Dimensions.get('window').width,
  fullHeight: Dimensions.get('window').height,
};

const Button = ({ children, onClick }) => {

  return (
    <TouchableOpacity
      testID={'button'}
      onPress={onClick}
      style={styles.button}
    >
      <Text style={styles.buttonText}>
        {children}
      </Text>
    </TouchableOpacity>
  );
}


Button.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
}

Button.defaultProps = {
  onClick: () => {},
}

export default Button;

const styles = StyleSheet.create({
  button:{
    display: 'flex',
    backgroundColor:'#000000',
    width: viewport.fullWidth* 0.80,
    padding: viewport.fullWidth * 0.02,
    alignItems:'center',
    justifyContent:'center',
    marginBottom:10,
  },
  buttonText:{
    color:'#fff',
    fontSize:20,
  }
});