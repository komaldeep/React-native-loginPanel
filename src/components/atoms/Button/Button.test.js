import React from 'react';
import findById from './../../../../test/find';
import { shallow, mount } from 'enzyme'
import Button from './Button';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const rendered = renderer.create(<Button> this is button </Button>).toJSON();
  expect(rendered).toMatchSnapshot();
});

test('onPress of button', () => {
  const component = renderer.create(<Button
    onClick={()=>{}}
  >
    BUTTON
  </Button>).toJSON();

  expect(findById(component, 'button')).toBeDefined()
});
