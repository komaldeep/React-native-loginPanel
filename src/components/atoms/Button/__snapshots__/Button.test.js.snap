// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "display": "flex",
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "padding": 15,
      "width": 600,
    }
  }
  testID="button"
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
      }
    }
  >
     this is button 
  </Text>
</View>
`;
